torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -621,6 +621,14 @@ RegisterOperators reg(
                    "Unsupported element type for tolist; only int, float and bool are supported");
              }
 
+             // Convert float Tensors to double since tolist is not supported
+             // for float Tensors.
+             if (t.scalar_type() == at::ScalarType::Float) {
+               TORCH_WARN(
```

Comment:
I don't even really know that a warning is necessary. If the user is asking for a `List[float]`, that's what they're going to get—we just perform the casting for them.

Reply:
agreed -- this is a non-lossy transform so it doesn't need a warning